extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/DevServicesProcessor.java

```diff
@@ -139,7 +151,16 @@ private StartResult startContainer(String connectionName, DevServicesConfig devS
 
         DockerImageName dockerImageName = DockerImageName.parse(devServicesConfig.imageName.orElse(REDIS_6_ALPINE))
                 .asCompatibleSubstituteFor(REDIS_6_ALPINE);
-        FixedPortRedisContainer redisContainer = new FixedPortRedisContainer(dockerImageName, devServicesConfig.port);
+
+        if (devServicesConfig.shared && launchMode == DEVELOPMENT) {
```

Comment:
Can this be extracted too?
Maybe not easily.

Reply:
I'll take a look at this as well, as it would further reduce the duplicated code.

Reply:
@cescoffier I see two possibilities how this can be extracted:

- in a smaller (simpler) scope we could extract the conditional logic for label-based container location (lines 155-160)
- in a bit broader scope we could extract the above __and__ also the other possibility, that is a creation of a new `Container` - which is done when the `shared` container is not used. (lines 155-172).

WDYT?

Reply:
I would go with the first one.

Reply:
@cescoffier I pushed the latest changes, generally, the `ContainerLocater` was made more "functional" by using `Optional` instead of bare `null`. This has shortened the code and additionally, it allowed to compose the method calls in a more fluent way.